client/src/controllers/CountController.ts

```diff
@@ -0,0 +1,64 @@
+import { Controller } from '@hotwired/stimulus';
+import { ngettext } from '../utils/gettext';
+
+export class CountController extends Controller<HTMLFormElement> {
+  static targets = ['total', 'label'];
+  static values = {
+    containerBody: { type: String, default: 'body' },
+    min: { type: Number, default: 0 },
+    target: {
+      type: String,
+      default:
+        '.error-message,.help-critical,[aria-invalid="true"], .error, .w-field--error',
+    },
+    total: Number,
+  };
+
+  declare minValue: number;
+  declare totalValue: number;
+  declare containerValue: string;
+  declare targetValue: string;
+
+  declare readonly hasTotalTarget: string;
+  declare readonly totalTarget: HTMLInputElement;
+  declare readonly totalTargets: HTMLInputElement[];
+
+  declare readonly hasLabelTarget: string;
+  declare readonly labelTarget: HTMLInputElement;
+  declare readonly labelTargets: HTMLInputElement[];
+
+  declare readonly hasActiveClass: string;
+  declare readonly activeClass: string;
+  declare readonly activeClasses: string[];
+
+  connect() {
+    this.count();
+  }
+
+  count() {
+    const containerElement = document.getElementsByTagName(
+      this.containerValue,
+    )[0];
+    const errorElements = containerElement.getElementsByClassName(
+      this.targetValue,
+    );
+    this.totalValue = errorElements.length;
```

Comment:
```suggestion
    const containerElement = document.querySelector(this.containerValue || 'body');
    this.totalValue = containerElement.querySelectorAll(
      this.targetValue,
    ).count;
```

Best to use `querySelector` and `querySelectorAll` as they work in a more general way compared to `getElementsByClassName`. For example, `getElementsByClassName` will not work if given `body` (as in all elements within body) but `querySelectorAll` will.